### PR TITLE
SHOP-1561 Final Mile changes after much Apollo churn

### PR DESCRIPTION
**Pull request by cantremember**
- some inline doc from observations
- throw any Errors coming back from the Apollo test client
- copied "final" createApolloTestClient pattern from 'server-core-test'

### CODE REVIEW

File: test/apollo/core.types.color.ts [hunk 1]
```diff
@@ -47,7 +47,10 @@ describe('the GraphQL Color TypeDefs', () => {
       `
     });
 
-    const { data } = res;
+    const { data, errors } = res;
+    if (errors?.length) {
+      throw errors[0];
+    }
```
Comment (cantremember) [inline test/apollo/core.types.color.ts:53]: i added this as boilerplate to all the Test Cases using an Apollo test client

general use of this pattern revealed that -- everything was being covered correctly!

File: test/apollo/core.types.void.ts [hunk 2]
```diff
@@ -44,24 +47,28 @@ describe('the GraphQL Void TypeDef', () => {
...
+      if (errors?.length) {
+        throw errors[0];
+      }
       expect(data).toMatchObject({
         voidQuery: null,
       });
     });
 
-    it('resolves null regardless of what the Resolver returns', async () => {
+    it('gets unhappy when its Resolver returns a non-nullable object', async () => {
```
Comment (cantremember) [inline test/apollo/core.types.void.ts:59]: ... and this is where throwing Errors coming out of the GraphQL response was revealing.  it was actually not behaving the Test Suite suggested

if a Developer intends to use the `Void` type, their Function had better return `null` or `undefined`

File: test/helpers/apollo.ts [hunk 3]
```diff
@@ -18,57 +18,70 @@ interface ITestSetupApollo {
...
-};
-
 const defaultMocks = {
     String: () => null
 };
 
-type StringOrAst = string | DocumentNode;
+//= = = = =
+//this code is taken from apollo-server-testing module which is deprecated
+//we implemented a custom 'client' wrapper matching the old `createTestClient` spec to minimize the impact during our upgrade
```
Comment (cantremember) [inline test/helpers/apollo.ts:27]: i copypasta'd the "final" version of this from 'server-core-test'

File: test/helpers/apollo.ts [hunk 4]
```diff
@@ -18,57 +18,70 @@ interface ITestSetupApollo {
...
+  query: (query: Query<ServerContext>) =>  Promise<FormattedExecutionResult<Record<string, unknown>, ObjMap<unknown>>>
+  mutate: (mutation: Mutation<ServerContext>) => Promise<FormattedExecutionResult<Record<string, unknown>, ObjMap<unknown>>>
 }
 
-export function createTestClient(apollo: ApolloServer<Context>, contextValue: Context): ApolloServerTestClient {
-  const graphqlResult = async ({ query, mutation, ...args }: Query | Mutation) => {
+//end of apollo-server-testing module copypasta
+//= = = = =
+
+function createApolloTestClient<ServerContext extends Context>(apollo: ApolloServer<ServerContext>, contextValue: ServerContext): ApolloServerTestClient<ServerContext> {
```
Comment (cantremember) [inline test/helpers/apollo.ts:64]: renamed, like in 'server-core-test'

File: test/helpers/apollo.ts [hunk 5]
```diff
@@ -18,57 +18,70 @@ interface ITestSetupApollo {
...
-      const derivedQuery = typeof operation === 'string' ? operation : print((operation as DocumentNode));
+      const derivedQuery = typeof operation === 'string' ? operation : print((operation as DocumentNode))
       const result = await apollo.executeOperation({ query: derivedQuery, ...args }, { contextValue: args.context || contextValue });
       const body = result.body as GraphQLResponseBodyKindSingle;
       assert.equal(body.kind, 'single');
       return body.singleResult;
     }
+    catch (err) {
+      return {
+        // conforming to `FormattedExecutionResult.errors`
```
Comment (cantremember) [inline test/helpers/apollo.ts:76]: behaves like 'server-core-test'

**Review by cantremember — COMMENTED** [review]
## annotated

most of this is polish, but the Test Suite changes were illuminating

**Review by vikas-mindpath — APPROVED** [review]
**LGTM**

**Pull request merged**